src/Core/Configurations/RuntimeConfigValidator.cs

```diff
@@ -1075,6 +1054,63 @@ public void ValidateRelationshipsInConfig(RuntimeConfig runtimeConfig, IMetadata
         }
     }
 
+    /// <summary>
+    /// Handles the validation of source and target fields as valid backing columns in the DB.
+    /// We put the logic for both source and target fields into a single function here for readability
+    /// when catching exceptions in the caller related to being unable to lookup the backing columns.
+    /// This happens when validation is called with a mal formed connection string.
```

Comment:
Why cant we skip such a validation when we know its a malformed connection string? 
`if (!ConfigValidationExceptions.Any(x => x.Message.StartsWith(DataApiBuilderException.CONNECTION_STRING_ERROR_MESSAGE)))` => cant we do this check?

Reply:
This can work, but i will need to change the validation we use for the connection string. Currently, that list of errors only contains the CONNECTION_STRING_ERROR_MESSAGE when the connection string is empty.


![image](https://github.com/Azure/data-api-builder/assets/93220300/a43d651f-75da-4263-87c7-d823d1b4dcd4)


So instead we will need better validation logic to throw those errors for any malformed connection string. I am adding that logic now.

Reply:
Yes, this should work. This check happens before we do any meta data validation, but it turns out that the DataSource validation that we do is only checking if the connection string is empty, not if it is invalid in any way.

![image](https://github.com/Azure/data-api-builder/assets/93220300/87ccfb61-0b93-4760-af45-5f30ce62385b)


But then later we add the exceptions from the meta data provider, and these include errors that are done by checking the validity of the actual connection to the backend DB, so we can use this check later on inside of the validation

![image](https://github.com/Azure/data-api-builder/assets/93220300/9db88120-ae1e-44f2-a697-258970907b77)


![image](https://github.com/Azure/data-api-builder/assets/93220300/f7b9be62-30ab-4985-b077-a59f72882659)

Reply:
To give a little more context on how this will ultimately be handled (new changes pending). The ValidateEntitiesMetadata function is validating entities in ways that both do and do not require metadata surrounding those entities from the backend. Rather than having to separately check for metadata initialization errors for each of this checks individually, we are going to split this function into 2 separate functions. One that doesn't rely on metadata from the backend, and one that does. We will then use the conditional above to make sure we dont attempt any checks that require this metadata in the case where we have those initialization errors that prevented that data from populating in the metadata provider, while still continuing with the function that validates information that doesnt depend on that meta data.